packages/migrate/migration.js

```diff
@@ -198,6 +201,47 @@ class Migration {
     return { interfaceAdapter, resolver, context, deployer };
   }
 
+  async deployAndLinkLogger(options, resolver) {
```

Comment:
Hm, @truffle/migrate seems like a weird place for this. Did you think of other options for where to put this logic?

Reply:
The way this works is that we have to link the logging library "behind-the-scenes" during the migration step. Unfortunately there's no way to have this outside of the migration step as far as I know.

@CruzMolina can chime in here if I got anything wrong.